src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -173,63 +173,51 @@ private IEnumerable<ITestCaseData> GetTestCasesFor(IMethodInfo method)
 
                         if (parms == null)
                         {
-                            // 3. An array was passed, it may be an object[]
-                            //    or possibly some other kind of array, which
-                            //    TestCaseSource can accept.
-                            var args = item as object[];
-                            if (args == null && item is Array)
-                            {
-                                Array array = item as Array;
-#if NETCF
-                                bool netcfOpenType = method.IsGenericMethodDefinition;
-#else
-                                bool netcfOpenType = false;
-#endif
-                                int numParameters = netcfOpenType ? array.Length : method.GetParameters().Length;
-                                if (array != null && array.Rank == 1 && array.Length == numParameters)
-                                {
-                                    // Array is something like int[] - convert it to
-                                    // an object[] for use as the argument array.
-                                    args = new object[array.Length];
-                                    for (int i = 0; i < array.Length; i++)
-                                        args[i] = array.GetValue(i);
-                                }
-                            }
+                            var args = new object[0];
 
-                            // Check again if we have an object[]
-                            if (args != null)
+                            var argsArray = item as Array;
+                            if (argsArray != null)
                             {
 #if NETCF
-                                if (method.IsGenericMethodDefinition)
+                                if(method.IsGenericMethodDefinition)
```

Comment:
We've removed NETCF from this repo, so when you merge latest and resolve conflicts, this code can be dropped.